akka-stream/src/main/scala/akka/stream/impl/fusing/Ops.scala

```diff
@@ -2351,8 +2351,16 @@ private[akka] final class StatefulMapConcat[In, Out](val f: () => In => Iterable
 
     override def onPush(): Unit =
       try {
-        currentIterator = plainFun(grab(in)).iterator
```

Comment:
I don't think this optimization is worth adding. You would normally return `Nil` if you know that it's empty and that is already using a single shared empty iterator instance.